corehq/util/workbook_reading/adapters/csv.py

```diff
@@ -0,0 +1,125 @@
+import ast
+from contextlib import contextmanager
+import datetime
+from dateutil.parser import parse
+import csv
+import os
+
+from corehq.util.workbook_reading import (
+    SpreadsheetFileEncrypted,
+    SpreadsheetFileInvalidError,
+    SpreadsheetFileNotFound,
+    Workbook,
+)
+from .raw_data import make_worksheet
+
+
+@contextmanager
+def open_csv_workbook(filename):
+    try:
+        if os.stat(filename).st_size <= 1:
+            raise SpreadsheetFileInvalidError('File is empty')
+        with open(filename, "r") as csv_file:
+            yield _CSVWorkbookAdaptor(csv_file).to_workbook()
+    except UnicodeDecodeError as error:
+        raise SpreadsheetFileEncrypted(str(error))
+    except csv.Error as error:
+        if str(error) == 'line contains NUL':
+            raise SpreadsheetFileEncrypted('Workbook is encrypted')
```

Comment:
This seems like a shaky way to detect encrypted file content. For example, there are things unrelated to encryption that could cause `UnicodeDecodeError`. I would expect an encrypted file to have some sort of header or other means of detecting that it is encrypted. Would it make sense to only raise `SpreadsheetFileEncrypted` when we know for sure the content is encrypted rather than guessing? Another option would be to make these error messages less specifically about encrypted content and more generically about unreadable file content.

Reply:
Good point. I think I'd prefer to leave the code less specific, and just tell the user that the content is unreadable. I'll try to update it to do that, and update the tests.

Reply:
**python-magic** is included in requirements.txt, and could be useful:
```python
>>> import magic
>>> magic.from_file('ascii.csv')
'ASCII text'
>>> magic.from_file('utf-8.csv')
'UTF-8 Unicode text'
>>> magic.from_file('encrypted.csv')
'data'
>>> magic.from_file('empty.csv')                                                                                         
'empty'
```